lib/api/controllers/readController.js

```diff
@@ -15,7 +15,16 @@ function ReadController (kuzzle) {
   this.search = function readSearch (requestObject) {
     var modifiedRequestObject = null;
 
-    return kuzzle.pluginsManager.trigger('data:beforeSearch', requestObject)
+    return new Promise((resolve, reject) => {
+      if (requestObject.index && typeof requestObject.index === 'string' && requestObject.index.split(',').length > 1) {
+        reject(new BadRequestError('Search on multiple indexes is not available.'));
```

Comment:
Since rejecting does not involve breaking the code flow, then in that case, the promise is rejected AND fulfilled.

Should be:
`return reject(new BadRequestError('Search on multiple indexes is not available.'));`